src/hotspot/share/services/heapDumper.cpp

```diff
@@ -988,7 +996,7 @@ void DumperSupport::dump_double(AbstractDumpWriter* writer, jdouble d) {
 }
 
 // dumps the raw value of the given field
-void DumperSupport::dump_field_value(AbstractDumpWriter* writer, char type, oop obj, int offset) {
+void DumperSupport::dump_field_value(AbstractDumpWriter* writer, char type, oop obj, int offset, bool redact) {
```

Comment:
I think it would be cleaner to specialize this method into two: `dump_field_value` and `dump_redacted_field_value`, and dispatch to it from outside. Likely saves branches.